transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/HttpDownload.java

```diff
@@ -148,22 +148,21 @@ private void writeTarget()
             }
             catch ( final IOException e )
             {
-                ConcreteResource resource = null;
-                if ( target.exists() )
+                closeQuietly( in );
+                closeQuietly( out );
+
+                ConcreteResource resource = target.getResource();
+                try
+                {
+                    target.delete();
+                }
+                catch ( IOException e1 )
                 {
-                    try
-                    {
-                        resource = target.getResource();
-                        target.delete();
-                    }
-                    catch ( IOException e1 )
-                    {
-                        throw new TransferException(
-                                        "Failed to write to local proxy store: {}\nOriginal URL: {}. Reason: {}", e1,
-                                        target, url, e1.getMessage() );
-                    }
+                    throw new TransferException( "Failed to delete target file: {}\nOriginal URL: {}. Reason: {}", e1,
```

Comment:
If you throw this TransferException without logging the IOException `e`, we lose the original cause of the problem. That will add days or weeks to any production problem that might happen in this code, and require us to fix & redeploy to production before we can even debug the problem.

Please log (to ERROR log level) `e1` and throw TransferException with the original IOException as the cause here.